src/awkward/_v2/_util.py

```diff
@@ -29,6 +29,8 @@
 kSliceNone = kMaxInt64 + 1  # for Slice::none()
 kMaxLevels = 48
 
+backend = {ak.nplike.Numpy: "cpu", ak.nplike.Cupy: "cuda"}
```

Comment:
This lookup table is going the opposite way of what you want: users give you a string and you need to return the singleton. Also, if the lookup dict (data) is defined here, the function that does the lookup (methods) should be here, too. That's the spirit of OOP, even if we're not going to the trouble of making actual objects.

So

```python
_backends = {
    "cpu": ak.nplike.Numpy,
    "cuda": ak.nplike.Cupy,
}

def regularize_backend(backend):
    if backend in _backends:
        return _backends[backend].instance()
    else:
        complain!
```

I realized while writing this that calling `ak.nplike.Cupy.instance()` might try to import `awkward_cuda_kernels`, so it should be delayed until the user is actually asking for it.

Reply:
I just realized that if you're using this internally, you'll have an `nplike` singleton, rather than a string naming it. Somewhere, you'll want a convenience function that passes through that `nplike`—maybe here, maybe not.